Resfeber/CoreData/Model/Trip+CoreDataProperties.swift

```diff
@@ -43,5 +43,21 @@ extension Trip {
 }
 
 extension Trip : Identifiable {
-
+    var eventsArray: [Event] {
+        let eventsArray = events?.allObjects as? [Event] ?? []
+        
+        /// Events sorted by date; events without a date are positioned at the end
+        return eventsArray.sorted { (x, y) -> Bool in
+            
+            if let xStartDate = x.startDate {
+                if let yStartDate = y.startDate {
+                    return xStartDate < yStartDate
+                } else {
+                    return true
+                }
+            }
+            
+            return y.startDate == nil
+        }
```

Comment:
```suggestion
        /// Events are sorted by start date in ascending order
        /// If an event does not have a start date, the end date is used
        /// If the dates are equal, events sorted by end date appear before events sorted by start date
        /// Events with no start or end date are positioned at the end
        return events.sorted { (a, b) -> Bool in
            if let aStartDate = a.startDate {
                if let bStartDate = b.startDate {
                    return aStartDate < bStartDate
                } else if let bEndDate = b.endDate {
                    return aStartDate < bEndDate
                } else {
                    return true
                }
            } else if let aEndDate = a.endDate {
                if let bStartDate = b.startDate {
                    return aEndDate <= bStartDate
                } else if let bEndDate = b.endDate {
                    return aEndDate < bEndDate
                } else {
                    return true
                }
            }
            return false
        }
```

Reply:
Implemented suggestion by @johnnydhicks to fall back on the end date if the start date is nil when sorting events.